src/sql-parser/src/parser.rs

```diff
@@ -2849,6 +2822,60 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn parse_alter_index(&mut self) -> Result<Statement<Raw>, ParserError> {
+        let if_exists = self.parse_if_exists()?;
+        let name = self.parse_object_name()?;
+
+        match self.expect_one_of_keywords(&[RESET, SET, RENAME]) {
```

Comment:
This should just use `?` to propagate the error to the caller

Reply:
you will have to be more specific. I don't think that I can combine `?` with a `match`

Reply:
Wanted to make sure my suggestions were sound, so here are all of them. lmk if there's anything here I can help explain.

```diff
diff --git a/src/sql-parser/src/parser.rs b/src/sql-parser/src/parser.rs
index 4eeb1228f..141e0e232 100644
--- a/src/sql-parser/src/parser.rs
+++ b/src/sql-parser/src/parser.rs
@@ -2826,54 +2826,47 @@ impl<'a> Parser<'a> {
         let if_exists = self.parse_if_exists()?;
         let name = self.parse_object_name()?;

-        match self.expect_one_of_keywords(&[RESET, SET, RENAME]) {
-            Ok(RESET) => {
+        Ok(match self.expect_one_of_keywords(&[RESET, SET, RENAME])? {
+            RESET => {
                 self.expect_token(&Token::LParen)?;
                 let reset_options = self.parse_comma_separated(Parser::parse_identifier)?;
                 self.expect_token(&Token::RParen)?;

-                return Ok(Statement::AlterIndex(AlterIndexStatement {
+                Statement::AlterIndex(AlterIndexStatement {
                     index_name: name,
                     if_exists,
                     action: AlterIndexAction::ResetOptions(reset_options),
-                }));
+                })
             }
-            Ok(SET) => {
+            SET => {
                 if self.parse_keyword(ENABLED) {
-                    return Ok(Statement::AlterIndex(AlterIndexStatement {
+                    Statement::AlterIndex(AlterIndexStatement {
                         index_name: name,
                         if_exists,
                         action: AlterIndexAction::Enable,
-                    }));
+                    })
                 } else {
                     let set_options = self.parse_with_options(true)?;
-                    return Ok(Statement::AlterIndex(AlterIndexStatement {
+                    Statement::AlterIndex(AlterIndexStatement {
                         index_name: name,
                         if_exists,
                         action: AlterIndexAction::SetOptions(set_options),
-                    }));
+                    })
                 }
             }
-            Ok(RENAME) => {
+            RENAME => {
                 self.expect_keyword(TO)?;
                 let to_item_name = self.parse_identifier()?;

-                return Ok(Statement::AlterObjectRename(AlterObjectRenameStatement {
+                Statement::AlterObjectRename(AlterObjectRenameStatement {
                     object_type: ObjectType::Index,
                     if_exists,
                     name,
                     to_item_name,
-                }));
-            }
-            Ok(_) => {
-                return self.expected(
-                    self.peek_prev_pos(),
-                    "RESET, SET, or RENAME",
-                    self.peek_token(),
-                )
+                })
             }
-            Err(e) => return Err(e),
-        };
+            _ => unreachable!(),
+        })
     }

     /// Parse a copy statement
```